com.unity.netcode.gameobjects/Runtime/Connection/NetworkConnectionManager.cs

```diff
@@ -104,9 +104,11 @@ internal void InvokeOnClientConnectedCallback(ulong clientId)
                     {
                         continue;
                     }
-
-                    peerClientIds[idx] = peerId;
-                    ++idx;
+                    if (peerClientIds.Length > idx)
```

Comment:
This assures if the server has not timed out prior to the client synchronizing that it doesn't exceed the allocated peer count.